README.md

```diff
@@ -54,6 +63,403 @@ npm install @dojo/shim
+const config: RouteConfig[] = [
+	{
+		path: 'foo',
+		outlet: 'root',
+		children: [
+			{
+				path: 'bar',
+				outlet: 'bar'
+			},
+			{
+				path: 'baz',
+				children: [
+					{
+						path: 'qux'
+					}
+				]
+			}
+		]
+	}
+]
+```
+
+That would register the following routes and outlets:
+
+| Route        | Outlet |
+| ------------ | ------ |
+|`/foo`        | `root` |
+|`/foo/bar`    | `bar`  |
+|`/foo/baz`    | `baz`  |
+|`/foo/baz/qux`| `qux`  |
+
+**Note:** If an `outlet` is not explicitly specified the `path` will be used.
+
+To actually register the configuration with a Dojo 2 router, simply pass the config object in the `constructor` options
+
+```ts
+const router = new Router({ config });
+```
+
+For routes that have either path parameters or query parameters it is possible to specify default parameters. These parameters are used as a fallback when generating a link from an outlet without specifying parameters or the params do not exist in the current route.
+
+```ts
+const config = [
+	{
+		path: 'foo/{foo}',
+		outlet: 'foo',
+		defaultParams: {
+			foo: 'bar'
+		}
+	}
+];
+
+// Using router.link to generate a link for an outlet 'foo', will use the default 'bar' value
+router.link('foo')
+```
+
+A default route can be specified using the optional configuration property `defaultRoute` which will be used if the current route doesn't match an existing. Note there can only be one default route configured otherwise an error will be thrown.
+
+#### Registering Additional Routes
+
+Additional routing configuration can be registered with a router instance, either from the root or by specifying an existing outlet name.
+
+```ts
+const additionalRouteConfig = [
+	{
+		path: 'extra',
+		outlet: 'extra'
+	}
+];
+
+// Will register the extra routes from the route
+router.register(additionalRouteConfig);
+
+// Will register the extra routes from `foo` outlet
+router.register(additionalRouteConfig, 'foo');
+```
+
+If the outlet is not found then an error is thrown.
+
+#### Router Context Injection
+
+To make the `router` instance available to the created outlets and other routing components (such as [`Link`](#link)) Routing leverages a core Dojo 2 widget-core concept [Injecting State](https://github.com/dojo/widget-core/blob/master/README.md#injecting-state). The custom injector `RouterInjector` needs to be defined in a registry available to the components for an known `key`.
+
+All routing components by default use the exported `RouterInjector#routerKey` as the key for the injected `router` context.
+
+```ts
+import { registry } from '@dojo/widget-core/d'
+import { RouterInjector, routerKey } from '@dojo/routing/RouterInjector';
+
+registry.define(routerKey, Injector(RouterInjector, router));
+```
+
+`RouterInjector` module exports a useful helper function, `registerRouterInjector` that combines the instantiation of a `Router` instance, registering route configuration and defining the `RouterInjector` and returns the `router` instance.
```

Comment:
"configuration and" -> "configuration, and" (I think the extra comma helps with clarity even though it is optional